nhs/data/filter.py

```diff
@@ -0,0 +1,27 @@
+from typing import List
+import polars as pl
+
+
+def filter_sa1_regions(lf: pl.LazyFrame, sa1_column: str = "SA1_CODE_2021", region_codes: List[str] = []) -> pl.LazyFrame:
+    """
+    Filters the LazyFrame to include only rows with specified SA1 area codes.
+
+    Parameters
+    ----------
+    lf : pl.LazyFrame
+        The LazyFrame containing SA1 region codes and data to be filtered.
+    SA1_column : str
+        The name of the column containing the SA1 area codes. Defaults to "SA1_CODE_2021".
+    region_codes : List[str]
+        A list of SA1 area codes to filter for.
+
+    Returns
+    -------
+    pl.LazyFrame
+        A LazyFrame containing only rows with the specified SA1 area codes.
+    """
+    if not region_codes:
+        return lf  # If no region codes are provided, return the original LazyFrame
```

Comment:
would be more accurate to have an empty `LazyFrame` if the list is empty, we can just remove this if statement and let `is_in()` do its thing

And maybe change `region_codes = []` to `region_codes` (i.e. remove default value `[]`) since the point of the function is to filter by these region codes.